### ISSUE TITLE
negative loss 

### ISSUE THREAD

**Issue by TianyuanYu** [cgraber/cvpr_dNRI#4]
Hi Colin,

Thanks for sharing the code of the excellent work. 

When I run the code on motion-118, I notice that the output val loss is negative value (-0.56...). This is caused by the reconstruction loss (nll_gaussian in dnri.py). The normalize_nll is defaultly set to be True so the negative constant is added with the reconstruction loss. 

It is not reasonable for me if any loss if negative. I thought it might be caused by different versions of pytorch. Just confirm whether you got negative loss in your result.

Thanks.

**Comment by TianyuanYu** [comment]
Hi,

I realised that the loss function is shifted by a negative constant. But I did not figure out why you add it. It seems that it should not affect the training procedure with or without this constant. I notice that theoretically the constant exists in formula (4). But I also notice that the constant is not used in NRI paper and other two forms of this loss (normalize_nll_per_var and else in nll_gaussian). Does the constant really matters? Could you share your thoughts here? Thanks very much.

**Comment by cgraber** [comment]
Hi Tianyuan, thanks for your interest! The constant represents a term in the Gaussian likelihood which does not depend on model parameters. You are correct in saying that it doesn't affect training: though choosing to include/exclude it changes the value of the loss, it does not change the gradient of model parameters with respect to the loss. Hence, some people choose to exclude it during computation (which the original NRI paper may have done). At some point, I clearly attempted to include the constant into the computation, but it seems like I did it incorrectly (my bad). 

I also obtain negative likelihood values on my end, so that's not due to a pytorch versioning issue.

**Comment by TianyuanYu** [comment]
Thanks for your reply. Very helpful.
